repo_policy_compliance/__init__.py

```diff
@@ -227,3 +234,78 @@ def collaborators(github_client: Github, repository_name: str) -> Report:
         )
 
     return Report(result=Result.PASS, reason=None)
+
+
+@inject_github_client
+def execute_job(
+    github_client: Github,
+    repository_name: str,
+    source_repository_name: str,
+    branch_name: str,
+    commit_sha: str,
+) -> Report:
+    """Check that the execution of the workflow for a SHA has been granted for a PR from a fork.
+
+    Args:
+        github_client: The client to be used for GitHub API interactions.
+        repository_name: The name of the repository to run the check on.
+        source_repository_name: The name of the repository that contains the source branch.
+        branch_name: The name of the branch that has the PR.
+        commit_sha: The SHA of the commit that the workflow run is on.
+
+    Returns:
+        Whether the workflow run has been approved for the commit SHA.
+    """
+    # Not from a forked repository
+    if repository_name == source_repository_name:
+        return Report(result=Result.PASS, reason=None)
+
+    # Retrieve PR for the branch
+    repository = github_client.get_repo(repository_name)
+    pulls = repository.get_pulls(state="open")
+    pull_for_branch = next((pull for pull in pulls if pull.head.ref == branch_name), None)
+    if not pull_for_branch:
+        return Report(result=Result.FAIL, reason=f"no open pull requests for branch {branch_name}")
+
+    # Retrieve comments on the PR
+    comments = pull_for_branch.get_issue_comments()
+    if not comments.totalCount:
+        return Report(
+            result=Result.FAIL,
+            reason=(
+                f"no comment found on PR - {EXECUTE_JOB_MESSAGE}, {branch_name=}, {commit_sha=} "
+                f"{pull_for_branch.number=}"
+            ),
+        )
+
+    # Check for authroization comment
+    authorization_string = f"{AUTHORIZATION_STRING_PREFIX} {commit_sha}"
+    authorization_comments = tuple(
+        comment for comment in comments if authorization_string in comment.body
```

Comment:
Should we exclude authorization_string in the quoted section of comments?

Reply:
The quoted section should contains at least a `">"` in the line.

Reply:
Should be fixed now, thanks